### ISSUE TITLE
The "curly" rule doesn't account for do-while statements

### ISSUE THREAD

**Issue by jamesallardice** [eslint/eslint#167]
When the `curly` option is set to `true`, the following does not issue a warning:

``` javascript
do
    foo();
while (bar);
```

The `curly` rule should require a block statement as the body of the statement:

``` javascript
do {
    foo();
} while (bar);
```
